packages/transformer/src/IModelTransformer.ts

```diff
@@ -1651,6 +1691,97 @@ export class IModelTransformer extends IModelExportHandler {
     }
   }
 
+  public convertHelmertToTransform(
+    helmert: Helmert2DWithZOffset | undefined
+  ): Transform {
+    if (!helmert) {
+      return Transform.createIdentity();
+    }
+
+    const rotationXY = Matrix3d.createRotationAroundAxisIndex(
+      AxisIndex.Z,
+      Angle.createDegrees(helmert?.rotDeg)
+    );
+    rotationXY.scaleColumnsInPlace(helmert.scale, helmert.scale, 1.0);
+    const translation = Vector3d.create(
+      helmert.translationX,
+      helmert.translationY,
+      helmert.translationZ
+    );
+    const helmertTransform = Transform.createRefs(translation, rotationXY);
+
+    return helmertTransform;
+  }
+
+  public calculateSpatialTransfromFromHelmertTransforms(
+    srcDb: IModelDb,
+    targetDb: IModel
+  ): Transform {
+    if (
+      srcDb.geographicCoordinateSystem?.horizontalCRS === undefined ||
+      srcDb.geographicCoordinateSystem?.verticalCRS === undefined
+    ) {
+      throw new IModelError(
+        IModelStatus.BadRequest,
+        "Source iModel does not have a geographic coordinate system defined."
+      );
+    }
+    if (
+      targetDb.geographicCoordinateSystem?.horizontalCRS === undefined ||
+      targetDb.geographicCoordinateSystem.verticalCRS === undefined
+    ) {
+      throw new IModelError(
+        IModelStatus.BadRequest,
+        "Target iModel does not have a geographic coordinate system defined."
+      );
+    }
+    if (
+      !srcDb.geographicCoordinateSystem.horizontalCRS.equals(
+        targetDb.geographicCoordinateSystem.horizontalCRS
+      ) ||
+      !srcDb.geographicCoordinateSystem.verticalCRS.equals(
+        targetDb.geographicCoordinateSystem.verticalCRS
+      )
+    ) {
+      throw new IModelError(
+        IModelStatus.MismatchGcs,
+        "Source and target geographic coordinate systems must match to calculate the spatial transform."
+      );
+    }
+
+    if (
+      srcDb.geographicCoordinateSystem.additionalTransform?.helmert2DWithZOffset
+        ?.scale !==
+      targetDb.geographicCoordinateSystem.additionalTransform
+        ?.helmert2DWithZOffset?.scale
+    ) {
+      throw new IModelError(
+        IModelStatus.MismatchGcs,
+        "Spatial transform is non linear. Source and target Helmert transforms must have the same scale to calculate a linear spatial transform."
```

Comment:
Is this true? I thought scale was a linear transformation.

Reply:
When the spatial transform ends up with some scale, it's no longer a rigid transformation.  That's perfectly fine for a Transform, but I guess the iModel transformer doesn't allow non-rigid Transforms.

Reply:
I corrected this, thanks for the clarification